tests/ci/libfuzzer_test_check.py

```diff
@@ -0,0 +1,224 @@
+        result.append(f"RUN_BY_HASH_NUM={run_by_hash_num}")
+        result.append(f"RUN_BY_HASH_TOTAL={run_by_hash_total}")
+
+    return result
+
+
+def get_run_command(
+    fuzzers_path: Path,
+    repo_path: Path,
+    result_path: Path,
+    kill_timeout: int,
+    additional_envs: List[str],
+    ci_logs_args: str,
+    image: DockerImage,
+) -> str:
+    additional_options = ["--hung-check"]
+    additional_options.append("--print-time")
+
+    additional_options_str = (
+        '-e ADDITIONAL_OPTIONS="' + " ".join(additional_options) + '"'
+    )
+
+    envs = [
+        f"-e MAX_RUN_TIME={int(0.9 * kill_timeout)}",
+        # a static link, don't use S3_URL or S3_DOWNLOAD
+        '-e S3_URL="https://s3.amazonaws.com/clickhouse-datasets"',
+    ]
+
+    envs += [f"-e {e}" for e in additional_envs]
+
+    env_str = " ".join(envs)
+
+    return (
+        f"docker run "
+        f"{ci_logs_args} "
+        f"--workdir=/fuzzers "
+        f"--volume={fuzzers_path}:/fuzzers "
+        f"--volume={repo_path}/tests:/usr/share/clickhouse-test "
+        f"--volume={result_path}:/test_output "
+        f"--cap-add=SYS_PTRACE {env_str} {additional_options_str} {image}"
+    )
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("check_name")
+    parser.add_argument("kill_timeout", type=int)
+    parser.add_argument(
+        "--validate-bugfix",
+        action="store_true",
+        help="Check that added tests failed on latest stable",
+    )
+    parser.add_argument(
+        "--post-commit-status",
+        default="commit_status",
+        choices=["commit_status", "file"],
+        help="Where to public post commit status",
+    )
+    return parser.parse_args()
+
+
+def docker_build_image(image_name: str, filepath: Path) -> DockerImage:
+    # context = filepath.parent
+    docker_image = DockerImage(image_name)
+    build_cmd = f"docker build --network=host -t {image_name} {filepath}"
+    logging.info("Will build image with cmd: '%s'", build_cmd)
+    subprocess.check_call(
+        build_cmd,
+        shell=True,
+    )
+    return docker_image
+
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+
+    stopwatch = Stopwatch()
+
+    temp_path = TEMP_PATH
+    repo_path = REPO_COPY
+    reports_path = REPORTS_PATH
+
+    args = parse_args()
+    check_name = args.check_name
+    kill_timeout = args.kill_timeout
+    validate_bugfix_check = args.validate_bugfix
+
+    run_changed_tests = validate_bugfix_check
+    gh = Github(get_best_robot_token(), per_page=100)
+
+    # For validate_bugfix_check we need up to date information about labels, so pr_event_from_api is used
+    pr_info = PRInfo(
+        need_changed_files=run_changed_tests, pr_event_from_api=validate_bugfix_check
+    )
+
+    commit = get_commit(gh, pr_info.sha)
+    atexit.register(update_mergeable_check, gh, pr_info, check_name)
+
+    if not Path(temp_path).exists():
+        os.makedirs(temp_path)
```

Comment:
```suggestion
    temp_path.mkdir(parents=True, exist_ok=True)
```

end every similar place too, please